src/utransport.rs

```diff
@@ -16,9 +16,23 @@ use std::hash::{Hash, Hasher};
 use std::ops::Deref;
 use std::sync::Arc;
 
-use crate::{UMessage, UStatus, UUri};
+use crate::{UCode, UMessage, UStatus, UUri};
 
-/// `UListener` is the uP-L1 interface that provides a means to create listeners which are registered to [`UTransport`]
+/// A factory for URIs representing this uEntity's resources.
+///
+/// Implementations may use arbitrary mechanisms to determine the information that
+/// is necessary for creating URIs, e.g. environment variables, configuration files etc.
+pub trait LocalUriProvider: Send + Sync {
```

Comment:
Double-check whether I have the right understanding. The UPTransport should have its own ways (no matter env, config, constructor...) to initialize the `authority`, `entity_id`, and `version`. Then we can take the benefit of `LocalUriProvider` after implementing the trait, right?

I only have `authority` in `up-transport-zenoh-rust` now. I will also support getting `entity_id` and `version` from users.

Reply:
You got it exactly right. Does this (still) make sense to you?

Reply:
Yes, make sense to me.